lib/libutee/tee_api.c

```diff
@@ -143,13 +162,28 @@ TEE_Result TEE_InvokeTACommand(TEE_TASessionHandle session,
 	TEE_Result res;
 	uint32_t ret_origin;
 	struct utee_params up;
-
+#ifdef CFG_TEE_BENCHMARK
+	uint32_t allocated = 0;
+	void *timebuf = NULL;
+
+	if (params) {
+		timebuf = TEE_Malloc(TEE_BENCH_RB_SIZE, TEE_MALLOC_FILL_ZERO);
+		if (!timebuf)
+			return TEE_ERROR_OUT_OF_MEMORY;
+		allocated = 1;
+		params[TEE_BENCH_DEF_PARAM].memref.buffer = timebuf;
+		params[TEE_BENCH_DEF_PARAM].memref.size = TEE_BENCH_RB_SIZE;
+	}
+#endif
 	__utee_from_param(&up, paramTypes, params);
 	res = utee_invoke_ta_command((uintptr_t)session,
 				      cancellationRequestTimeout,
 				      commandID, &up, &ret_origin);
 	__utee_to_param(params, NULL, &up);
-
+#ifdef CFG_TEE_BENCHMARK
+	if (allocated)
```

Comment:
Right, anyway we will end up with `raw_free` call in both cases (where is a check for NULL). 
My last question was connected to why do we have different implementations of tee_user_mem_free in in [tee_user_mem.c](https://github.com/OP-TEE/optee_os/blob/master/lib/libutee/tee_user_mem.c)
Seems to me that code inside `CFG_NO_USER_MALLOC_GARBAGE`  section just isn't used anymore.